sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -123,17 +128,32 @@ export function convertResponseForQueryBatch(
       })
       ?.map((response: GeneratedBatchQueryResponse) => ({
         id: response.id,
-        status: response.status,
+        visualization: (JSON.parse(response.body as any) as BatchQueryResults).render,
```

Comment:
lots of points of failure here - lot of chances for null / undefined - are there any scenarios where this would throw at runtime and we should be defensive? 

If we think there's no chance of this data missing, or unparsable then maybe we should validate that and throw a better error message? 

If we think it's totally valid that theses fields are missing maybe we should be defensive / allow nulls?

Reply:
yes if these are undefined in the generated response, we allowed undefined as a value returned to user